### PR TITLE
feat(pfd): mda/mdh indication

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #[issue_no]

## Summary of Changes

Add MDA/MDH bug on the altitude tape.
Fix a couple of bugs, including:
- alt tape limits being too big for selected alt etc.
- digital alt value flashing amber briefly when MDA/MDH is changed.

## References

Docs... and
image
image
Selected alt coming onto scale:
image

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Check that MDA bug shows up at the right altitude in QNH mode, and MDH bug shows up at the right altitude in QFE mode. Check that selected alt appears/disappears at the right time going on/off-scale.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Review by 2hwk — CHANGES_REQUESTED** [review]

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js [hunk 1]
```diff
@@ -1525,6 +1533,45 @@ class FMCMainDisplay extends BaseAirliners {
...
+                landingElevation = isFinite(ele) ? ele : undefined;
+            }
+        }
+
+        if (this.landingElevation !== landingElevation) {
+            this.landingElevation = landingElevation;
+
+            const ssm = landingElevation !== undefined ? Arinc429Word.SignStatusMatrix.NormalOperation : Arinc429Word.SignStatusMatrix.NoComputedData;
+
+            this.setBnrArincSimVar('LANDING_ELEVATION', landingElevation ? landingElevation : 0, ssm, 14, 16384, -2048);
```
[the remark this answers is not in the document]
Comment (tracernz) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js:1556]: Limits for this value (from ref sources).

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js [hunk 2]
```diff
@@ -1525,6 +1533,45 @@ class FMCMainDisplay extends BaseAirliners {
         }
     }
 
+    async updateLandingElevation() {
+        let landingElevation;
+
+        /** @type {OneWayRunway} */
+        const runway = this.flightPlanManager.getDestinationRunway(FlightPlans.Active);
+        if (runway) {
+            landingElevation = runway.thresholdElevation / 0.3048;
```
[the remark this answers is not in the document]
Comment (tracernz) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js:1542]: You don't know the definition of a foot off-hand? :P

File: src/instruments/src/PFD/AltitudeIndicator.tsx [hunk 3]
```diff
@@ -106,6 +107,99 @@ class RadioAltIndicator extends DisplayComponent<{ bus: EventBus, filteredRadioA
...
+            this.updateIndication();
+        });
+
+        sub.on('altitudeAr').withArinc429Precision(0).handle((a) => {
+            // TODO filtered alt
+            this.altitude = a.value;
+            this.updateIndication();
+        });
+
+        sub.on('mda').handle((mda) => {
```
Comment (Saschl) [inline src/instruments/src/PFD/AltitudeIndicator.tsx:183]: I don't think that values changes very often. Consider using `.whenChanged()`

Comment (tracernz) [reply src/instruments/src/PFD/AltitudeIndicator.tsx]: Good catch! 😀

**Review by 2hwk — APPROVED** [review]

File: src/instruments/src/PFD/shared/ArincValueProvider.ts [hunk 4]
```diff
@@ -148,4 +188,17 @@ export class ArincValueProvider {
 
         publisher.pub('chosenRa', chosenRadioAltitude);
     }
+
+    private determineAndPublishChosenLandingElevation(publisher: Publisher<Arinc429Values>) {
+        const useOpposite = this.ownLandingElevation.isFailureWarning()
+            && this.ownLandingElevation.isNoComputedData()
+            && !this.oppLandingElevation.isFailureWarning()
+            && !this.oppLandingElevation.isNoComputedData();
```
Comment (lukecologne) [inline src/instruments/src/PFD/shared/ArincValueProvider.ts:196]: Looks like this is always false? Probably meant
```suggestion
        const useOpposite = (this.ownLandingElevation.isFailureWarning()
            || this.ownLandingElevation.isNoComputedData())
            && !this.oppLandingElevation.isFailureWarning()
            && !this.oppLandingElevation.isNoComputedData();
```

**Review by lukecologne — APPROVED** [review]
Logic of the PFD seems good, just one little thing I noticed, which shouldn't be too relevant right now with only one FMS.

File: src/instruments/src/PFD/shared/ArincValueProvider.ts (hunk 4 above)
Comment (tracernz) [reply src/instruments/src/PFD/shared/ArincValueProvider.ts:196]: Oof, brilliant catch. 👍

**Comment by 2hwk** [comment]
QA Report

Discord: 2Cas#1022
Object of testing: #7401
Tier of Testing: 1
Date: 27/08/2022

Testing Process: 
- [x] Check that MDA bug shows up at the right altitude in QNH mode, and MDH bug shows up at the right altitude in QFE mode. 
- [x] Check that selected alt appears/disappears at the right time going on/off-scale.
Negatives: none

Testing Results: Passed

Conclusions: LGTM

**Pull request merged**